File: src/models/output/operation_cost.jl
function gen_operation_cost_df(m)

    x_fuel = m[:x_fuel]
    x_CO2_flue = m[:x_CO2_flue]
    x_power_net = m[:x_power_net]
    x_cost_NGCC_VOM = m[:x_cost_NGCC_VOM]

    vars = [
        "fuel_cost", 
        "CO2_penalty",
        "power_profit",
        "x_cost_NGCC_VOM",
    ]

    df_cost = DataFrame(map(generate_row_map, vars))

    for i in set_hour
        fuel_cost = cost_NG * value(x_fuel[i])
        CO2_penalty = CO2_CREDIT * value(x_CO2_flue[i])
        power_profit = power_price[i] * value(x_power_net[i])
        x_cost_NGCC_VOM_t = value(x_cost_NGCC_VOM[i])
        push!(df_cost, (fuel_cost, CO2_penalty, power_profit, x_cost_NGCC_VOM_t))
    end

    return df_cost
end

File: src/pyomo_model/output/operation_cost.py
"""
Output module
Pengfei Cheng

Output DAC operation-related results.
"""

import pandas as pd
from pyomo.environ import value
from src.pyomo_model.params import *

def gen_operation_cost_df(m, set_hour, power_price, cost_NG):

    cols = [
        "profit_power",
        "credit_CO2",
        "cost_fuel",
        "cost_CO2_TS",
        "cost_NGCC_VOM",
        "cost_PCC_VOM",
        "cost_DAC_VOM",
        "cost_PCC_compr_VOM",
        "cost_DAC_compr_VOM"
    ]

    dfs = {}

    for s in set_scenario:
        df = pd.DataFrame(columns=cols)
        for i in set_hour:
            profit_power = power_price[i] * value(m.x_power_net[i, s])
            credit_CO2 = unit_CO2_price[s] * value(m.x_CO2_cap_total[i, s])
            cost_fuel = cost_NG * value(m.x_fuel[i, s])
            cost_CO2_TS = a_cost_CO2_TS * value(m.x_CO2_compress[i, s])
            cost_NGCC_VOM = value(m.x_cost_NGCC_VOM[i, s])
            cost_PCC_VOM = value(m.x_cost_PCC_VOM[i, s])
            cost_DAC_VOM = value(m.x_cost_DAC_VOM[i, s])
            cost_PCC_compr_VOM = value(m.x_cost_PCC_compr_VOM[i, s])
            cost_DAC_compr_VOM = value(m.x_cost_DAC_compr_VOM[i, s])
            res = [profit_power, credit_CO2, cost_fuel, cost_CO2_TS, cost_NGCC_VOM, cost_PCC_VOM, cost_DAC_VOM, cost_PCC_compr_VOM, cost_DAC_compr_VOM]
            df.loc[len(df)] = res
        dfs[s] = df

    return dfs